notebooks/geoserver.ipynb

```diff
@@ -0,0 +1,589 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #37.    GEOSERVER_SECURED_SERVICE = "geoserver-secured"
</pre><p>Need to check with https://github.com/bird-house/birdhouse-deploy/pull/348, as<code>geoserver-secured</code> will be replaced very soon by simply <code>geoserver</code>.</p><p> </p><p>I think what would be best is to remove the <code>/twitcher/ows/proxy</code>portion from URLs and use only <code>/geoserver</code>  (as in the <code>/geoserver/rest</code>  endpoint used right after).</p><p>This way, the logic of which location (if it maps to <code>geoserver</code> or <code>geoserver-secured</code>) will be irrelevant from this script. </p><p> </p><p><code>GEOSERVER_SECURED_SERVICE</code> will only be needed temporarily until the service on Magpie side is renamed accordingly with https://github.com/bird-house/birdhouse-deploy/pull/348.</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/125/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Reply:
<p>If the url changes from using <code>geoserver-secured</code> to <code>geoserver</code>, does that mean I have to also change the Magpie service name too from <code>geoserver-secured</code> to <code>geoserver</code>? If so, I get a conflict with an already existing <code>geoserver</code> wfs-type service with the current birdhouse setup. </p>

Reply:
Yes. Service named `geoserver` (of same Magpie `type: geoserver`) now includes all capabilities to combine REST API, WFS, WMS, WPS permissions under a common service. You should be able to perform the same requests to set up the service/resource permissions as before. I would expect the service itself to be already defined in the stack, so the notebook shouldn't worry about creating the service itself.

Reply:
<p>What I am refering to is this <a href="https://github.com/bird-house/birdhouse-deploy/blob/4884f82cf3913b65d92421df24a777169f9bec08/birdhouse/config/geoserver/config/magpie/providers.cfg.template#L10-L16" target="_blank">code</a> on <code>birdhouse-deploy</code> .</p><p>This creates a service named <code>geoserver</code> of type <code>wfs</code> (instead of the type <code>geoserver</code> that we want).</p><p>So I will get a conflict on my test if I try to create or get a service named <code>geoserver</code> of type <code>geoserver</code>.</p><p> </p><p>Should I update the referred code to use <code>geoserver</code> type/sync_type instead of <code>wfs</code>? I am unsure about what this provider definition was used for initially.</p>

Reply:
Yes. We're talking about the same service. 
This is the updated file in the pending PR with the change I was refering to: 
https://github.com/bird-house/birdhouse-deploy/blob/geoserver-behind-twitcher/birdhouse/config/geoserver/config/magpie/providers.cfg

In your tests, you should not even try to create the service. The notebook should assume that it already exists.
If the service doesn't exist, then the birdhouse-deploy configuration would be wrong altogether.

Reply:
<p>Alright, I made an env variable for the GEOSERVER_SERVICE, that uses <code>geoserver</code> by default. So, for now, we can still use the <code>geoserver-secured</code> with the env variable until the pending PR removes the <code>geoserver-secured</code> service officially.    </p>